core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -705,25 +709,36 @@ class KafkaApis(val requestChannel: RequestChannel,
           metadataRequest.topics.asScala.toSet
       }
 
-    var (authorizedTopics, unauthorizedTopics) =
+    var (authorizedTopics, unauthorizedForDescribeTopics) =
       topics.partition(topic => authorize(request.session, Describe, new Resource(Topic, topic)))
 
+    var unauthorizedForCreateTopics = Set[String]()
+      
     if (authorizedTopics.nonEmpty) {
       val nonExistingTopics = metadataCache.getNonExistingTopics(authorizedTopics)
       if (config.autoCreateTopicsEnable && nonExistingTopics.nonEmpty) {
         authorizer.foreach { az =>
           if (!az.authorize(request.session, Create, Resource.ClusterResource)) {
             authorizedTopics --= nonExistingTopics
-            unauthorizedTopics ++= nonExistingTopics
+            unauthorizedForCreateTopics ++= nonExistingTopics
           }
         }
       }
     }
 
-    val unauthorizedTopicMetadata = unauthorizedTopics.map(topic =>
+    val uncreatableTopicMetadata = unauthorizedForCreateTopics.map(topic =>
```

Comment:
Would it make sense to change this to `unauthorizedForCreateTopicMetadata`? Similarly, we could rename the variable below to `unauthorizedForDescribeTopicMetadata`.